### DISCUSSION TITLE
是否可以允许 latex 公式左右侧含有空格？

### DISCUSSION THREAD

**Discussion by Mr-Python-in-China** [mdit-plugins/mdit-plugins#32] [Q&A] [answered]
e.g.
```markdown
# Theatre Square

## 题面翻译

用 $ a \times a$ 的石板覆盖 $n \times m $ 的长方形广场，允许石板覆盖的区域超出广场，不允许打破石板，石板的两侧应平行于广场两侧，要求覆盖完广场所需的石板数量最少是多少。

感谢 @Loner_Knowledge  贡献翻译。

## 题目描述

Theatre Square in the capital city of Berland has a rectangular shape with the size $ n×m $ meters. On the occasion of the city's anniversary, a decision was taken to pave the Square with square granite flagstones. Each flagstone is of the size $ a×a $ .

What is the least number of flagstones needed to pave the Square? It's allowed to cover the surface larger than the Theatre Square, but the Square has to be covered. It's not allowed to break the flagstones. The sides of flagstones should be parallel to the sides of the Square.

## 输入格式

The input contains three positive integer numbers in the first line: $ n,m $ and $ a $ ( $ 1\leq n,m,a\leq10^{9} $ ).

## 输出格式

Write the needed number of flagstones.
```
---
**我希望的渲染效果：**
![image](https://github.com/mdit-plugins/mdit-plugins/assets/89737170/7e207b5f-2a46-4280-8022-d7240cfe7dc9)
---
**不过实际上的：**
![image](https://github.com/mdit-plugins/mdit-plugins/assets/89737170/33d08f73-87c0-494c-a14b-f594014a94bc)

这是由于 latex 公式左右出现空格导致的。
如果我想让他同样可以解析这样有空格的公式，应该怎么修改代码？

**Comment by Mister-Hope** [comment]
我们不会考虑这点，因为这会导致正常的美元符号的使用出现问题，而且极容易出现错误的匹配（解析了公式之间的文字把公式原样输出）。

另外，当公式本身出现美元符号时，这会使得 `$` 的意义难以被界定。

https://github.com/mdit-plugins/mdit-plugins/blob/18ddb0f5b0dbd3232869e3b5a821bee019acf398/packages/tex/__tests__/tex.spec.ts#L37-L50

你的请求违背了插件原本的设计理念，如果你有大量上述不标准的TeX代码，考虑使用其他插件。

**Comment by Mister-Hope** [comment, marked as answer]
看了一下，加个选项不难，0.6.5 增加了 allowInlineWithSpace 选项，但是它会破坏 $ 正常的渲染，自行斟酌使用。

https://github.com/mdit-plugins/mdit-plugins/blob/ab63c7abd96b087893695e4b7bd76349c684625f/packages/tex/__tests__/tex.spec.ts#L85-L100
